client/my-sites/plugins/controller-logged-out.js

```diff
@@ -0,0 +1,75 @@
+import { BASE_STALE_TIME } from 'calypso/data/marketplace/constants';
+import {
+	getFetchWPCOMFeaturedPlugins,
+	getFetchWPCOMPlugins,
+} from 'calypso/data/marketplace/use-wpcom-plugins-query';
+import { getFetchWPORGPlugins } from 'calypso/data/marketplace/use-wporg-plugin-query';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { getProductsList } from 'calypso/state/products-list/selectors';
+
+function getProps( context ) {
+	const props = {
+		path: context.path,
+		locale: context.lang,
+		tag: '',
+	};
+	return props;
+}
+
+function prefetchPluginsData( queryClient, fetchParams, infinite ) {
+	const queryType = infinite ? 'fetchInfiniteQuery' : 'fetchQuery';
+
+	return queryClient[ queryType ]( ...fetchParams, {
+		enabled: true,
+		staleTime: BASE_STALE_TIME,
+		refetchOnMount: false,
+	} ).catch( () => {} );
```

Comment:
`QueryClient` has special methods for prefetching: `queryClient.prefetchQuery` and `queryClient.prefetchInfiniteQuery`. The params are exactly the same, but the return value is `Promise< void >`: doesn't return the data, and also never fails, so the `.catch` handler can be removed.

The `enable: true` flag is redundant, as the query is enabled by default. Also, the `refetchOnMount` is not needed? Here we are not in React at all, we're calling the OOP API of `QueryClient` outside React. The `refetch*` params don't make sense here: the fetch is done only once, there's no React component that would live for an extended period of time and do refetches when needed.

Reply:
Thanks for the suggestions. I removed the unnecessary bits.
I can't remember why I thought one of the  `prefetch*` calls didn't work. I switched now and they seem to work fine.